src/app/shared/components/template/services/template-calc.service.ts

```diff
@@ -119,6 +119,31 @@ const CALC_FUNCTIONS: IFunctionHashmap = {
     }
   },
 
+  /**
+   * Mutate an array of arrays (expects a partition of a set): look for the sub-array
+   * containing one value and merge it with the sub-array containing another
+   * @param partition e.g. [[1, 2], [3], [4, 5, 6]]
+   * @param value1 e.g. 3
+   * @param value2 e.g. 5
+   * @returns A new partition, e.g. [[1, 2], [3, 4, 5, 6]]
+   */
+  merge_sub_arrays: (partition: any[][] | string, value1: string, value2: string) => {
```

Comment:
I don't think this name is very clear (I would expect merging the arrays to concatenate all the child entries into a single entry. If specific function just for one temp context I think probably call it something like `plh_merge_families` or something similarly explicit.

I would also recommend putting the function in its own file and adding a lightweight spec test

Reply:
I've pushed potential changes to #2127